project/src/components/offer-card/offer-card.jsx

```diff
@@ -1,26 +1,90 @@
+/* eslint-disable no-console */
 import React from 'react';
-import { Link } from 'react-router-dom';
+import { Link, useRouteMatch } from 'react-router-dom';
 import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { createPercent } from '../../utils/utils';
+import { Types } from '../../mocs/offers';
+import OfferCardProp from './offer-card.prop';
+import { AppRoutes } from '../../const';
+
+const classNamesByPath = {
+  [AppRoutes.ROOT]: {
+    article: 'cities__place-card',
+    image: 'cities__image-wrapper',
+    info: '',
+  },
+  [AppRoutes.FAVORITES]: {
+    article: 'favorites__card',
+    image: 'favorites__image-wrapper',
+    info: 'favorites__card-info',
+  },
+  [AppRoutes.OFFER_DETAILS]: {
+    article: 'near-places__card',
+    image: 'near-places__image-wrapper',
+    info: '',
+  },
+};
+
+export default function OfferCard({ offer, handlerOnMouseEnter }) {
+  const {
+    id,
+    price,
+    rating,
+    isFavorite,
+    title,
+    type,
+    isPremium,
+    previewImage,
+  } = offer;
+
+  const PictureSize = {
+    width: 260,
+    height: 200,
+  };
+
+  const { path } = useRouteMatch();
+  if (path === AppRoutes.FAVORITES) {
+    PictureSize.width = 150;
+    PictureSize.height = 110;
+  }
```

Comment:
Рассмотри такой вариант
```suggestion
  const isFavoriteView = path === AppRoutes.FAVORITES;
  const PictureSize = {    
    WIDTH: isFavoriteView ? 150 : 260,
    HEIGHT: isFavoriteView ? 110 : 200
  };
```